editor/plugins/tiles/atlas_merging_dialog.cpp

```diff
@@ -95,7 +95,9 @@ void AtlasMergingDialog::_generate_merged(Vector<Ref<TileSetAtlasSource>> p_atla
 			}
 		}
 
+		merged->set_name(p_atlas_sources[0]->get_name());
 		merged->set_texture(ImageTexture::create_from_image(output_image));
+		merged->set_texture_region_size(new_texture_region_size);
```

Comment:
I looked at the code and this line seems enough to fix the crash. The problem was that the region size was 0 and no tiles were created.